### PR TITLE
Handle Alt+Click for terminal links. Fixes #30761

### LINKED ISSUE #30761: Follow link in terminal results in save dialog
- VSCode Version: 1.14.1
- OS Version: macOS Sierra 10.12.5

Steps to Reproduce:

1. settings: `"editor.multiCursorModifier": "ctrlCmd"`
2. open integrated terminal
3. run `echo https://github.com`
4. alt+click on the link
5. save file dialog appears with filename `download.html`

screen shot 2017-07-15 at 14 15 27

Expected behaviour: it opens link in a browser

> Also the shortcut used to be CMD+click and not ALT+click. I couldn't find where to change it back :-(

This is actually caused by `editor.multiCursorModifier` settings. I was not aware of that when I created issue.

Reproduces without extensions: Yes

### PR DESCRIPTION
**Pull request by cleidigh**
@Tyriar 
- See #30761 for discussion
- Added hypertext link handling to bypass default electron link handling for Alt+Click (was opening save dialog)
- Tested on Windows and OS X
- The validate links should provide the error handling before the opener calls

Fixes #30761

### CODE REVIEW

File: src/vs/workbench/parts/terminal/electron-browser/terminalLinkHandler.ts [hunk 1]
```diff
@@ -72,7 +72,12 @@ export class TerminalLinkHandler {
 		const baseLocalLinkClause = _platform === platform.Platform.Windows ? winLocalLinkClause : unixLocalLinkClause;
 		// Append line and column number regex
 		this._localLinkPattern = new RegExp(`${baseLocalLinkClause}(${lineAndColumnClause})`);
-		this._xterm.setHypertextLinkHandler(this._wrapLinkHandler(() => true));
+
+		this._xterm.setHypertextLinkHandler(this._wrapLinkHandler(uri => {
+			this._handleHypertextLink(uri);
+			return;
```
Comment (Tyriar) [inline src/vs/workbench/parts/terminal/electron-browser/terminalLinkHandler.ts]: Let's remove this `return`, plus the one in `registerLocalLinkHandler`. Not sure why I put them there 😅

File: src/vs/workbench/parts/terminal/electron-browser/terminalLinkHandler.ts [hunk 2]
```diff
@@ -164,6 +170,12 @@ export class TerminalLinkHandler {
 		callback(true);
 	}
 
+	private _handleHypertextLink(url: string) {
```
Comment (Tyriar) [inline src/vs/workbench/parts/terminal/electron-browser/terminalLinkHandler.ts]: Add `: void` return type

File: src/vs/workbench/parts/terminal/electron-browser/terminalLinkHandler.ts [hunk 3]
```diff
@@ -164,6 +170,12 @@ export class TerminalLinkHandler {
...
+		let uri = Uri.parse(url);
+		this._openerService.open(uri);
+		return;
```
Comment (Tyriar) [inline src/vs/workbench/parts/terminal/electron-browser/terminalLinkHandler.ts]: Remove

**Review by Tyriar — CHANGES_REQUESTED** [review]
Looks great! Just a few minor comments 😃

**Comment by cleidigh** [comment]
@Tyriar 
Drat, I knew the return types were not quite clear!
Will fix up tomorrow.

File: src/vs/workbench/parts/terminal/electron-browser/terminalLinkHandler.ts (hunk 1 above)
Comment (cleidigh) [inline src/vs/workbench/parts/terminal/electron-browser/terminalLinkHandler.ts]: okay
anonymous function with void return, explicit return is redundant/unnecessary

**Review by Tyriar — APPROVED** [review]

**Pull request merged**